perl/html-option-list.1
.\" Automatically generated by Pod::Man 2.25 (Pod::Simple 3.20)
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  \*(C+ will
.\" give a nicer C++.  Capital omega is used to do unbreakable dashes and
.\" therefore won't be available.  \*(C` and \*(C' expand to `' in nroff,
.\" nothing in troff, for use with C<>.
.tr \(*W-
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
'br\}
.\"
.\" Escape single quotes in literal strings from groff's Unicode transform.
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\"
.\" If the F register is turned on, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.SS), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.ie \nF \{\
.    de IX
.    tm Index:\\$1\t\\n%\t"\\$2"
..
.    nr % 0
.    rr F
.\}
.el \{\
.    de IX
..
.\}
.\"
.\" Accent mark definitions (@(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2).
.\" Fear.  Run.  Save yourself.  No user-serviceable parts.
.    \" fudge factors for nroff and troff
.if n \{\
.    ds #H 0
.    ds #V .8m
.    ds #F .3m
.    ds #[ \f1
.    ds #] \fP
.\}
.if t \{\
.    ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.    ds #V .6m
.    ds #F 0
.    ds #[ \&
.    ds #] \&
.\}
.    \" simple accents for nroff and troff
.if n \{\
.    ds ' \&
.    ds ` \&
.    ds ^ \&
.    ds , \&
.    ds ~ ~
.    ds /
.\}
.if t \{\
.    ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.    ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.    ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.    ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.    ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.    ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.\}
.    \" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.    \" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.    \" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.    ds : e
.    ds 8 ss
.    ds o a
.    ds d- d\h'-1'\(ga
.    ds D- D\h'-1'\(hy
.    ds th \o'bp'
.    ds Th \o'LP'
.    ds ae ae
.    ds Ae AE
.\}
.rm #[ #] #H #V #F C
.\" ========================================================================
.\"
.IX Title "HTML-OPTION-LIST 1"
.TH HTML-OPTION-LIST 1 "2015-06-08" "perl v5.16.2" "User Contributed Perl Documentation"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.if n .ad l
.nh
.SH "NAME"
html\-option\-list \- Generate a list of HTML <option> elements
.SH "SYNOPSIS"
.IX Header "SYNOPSIS"
\&\fBhtml-option-list\fR [\fB\-w\fR \fIlength\fR] [\fB\-b\fR \fIstring\fR] [\fB\-a\fR \fIstring\fR] [\fB\-n\fR] [\fB\-v\fR] [\fB\-i\fR \fItabs\fR] [\fB\-\-width\fR \fIlength\fR] [\fB\-\-before\fR \fIstring\fR] [\fB\-\-after\fR \fIstring\fR] [\fB\-\-newline\fR] [\fB\-\-new\-line\fR] [\fB\-\-indent\fR \fItabs\fR] [\fB\-\-verbatim\fR]
.SH "DESCRIPTION"
.IX Header "DESCRIPTION"
Generate a list of \s-1HTML\s0 <option> elements from a sequence of arguments, typically numeric.
.PP
Yes, that's really all.
.SH "OPTIONS"
.IX Header "OPTIONS"
.IP "\fB\-w\fR \fIlength\fR, \fB\-\-width\fR \fIlength\fR" 6
.IX Item "-w length, --width length"
Pad the displayed value with leading zeroes to ensure it satisfies a minimum length.
.IP "\fB\-b\fR \fIstring\fR, \fB\-\-before\fR \fIstring\fR" 6
.IX Item "-b string, --before string"
Prepend each <option> element with \fIstring\fR.
.IP "\fB\-a\fR \fIstring\fR, \fB\-\-after\fR \fIstring\fR" 6
.IX Item "-a string, --after string"
Append \fIstring\fR to each <option> element.
.IP "\fB\-n\fR, \fB\-\-newline\fR, \fB\-\-new\-line\fR" 6
.IX Item "-n, --newline, --new-line"
Append a line-break character (\fIU+000A\fR) after each <option>, as though \fB\-\-after\fR=$'\en' were entered in the arguments. Overrides \fB\-\-after\fR.
.IP "\fB\-i\fR \fInum\fR, \fB\-\-indent\fR \fInum\fR" 6
.IX Item "-i num, --indent num"
Indent each <option> with \fInum\fR tabs. Overrides \fB\-\-before\fR.
.IP "\fB\-v\fR, \fB\-\-verbatim\fR" 6
.IX Item "-v, --verbatim"
Pad numeric values with zeroes, but don't use the unpadded value as a \fIvalue\fR attribute in the <option> element.
.SH "EXAMPLES"
.IX Header "EXAMPLES"
Generate a list of numbers representing the days of the month, with leading zeroes affixed where necessary:
.PP
.Vb 1
\&        html\-option\-list \-w 2 {1..31}
.Ve
.PP
Do the same as above, but \*(L"pretty-print\*(R" it so it's not crapped on one frikkin' long line:
.PP
.Vb 1
\&        html\-option\-list \-w 2 \-\-newline \-\-indent=1 {1..31}
.Ve


perl/html-option-list.pl
#!/usr/bin/env perl


=head1 NAME

B<html-option-list> - Generate a list of HTML <option> elements


=head1 SYNOPSIS

B<html-option-list> [B<-w> I<length>] [B<-b> I<string>] [B<-a> I<string>] [B<-n>] [B<-v>] [B<-i> I<tabs>] [B<--width> I<length>] [B<--before> I<string>] [B<--after> I<string>] [B<--newline>] [B<--new-line>] [B<--indent> I<tabs>] [B<--verbatim>]


=head1 DESCRIPTION

Generate a list of HTML <option> elements from a sequence of arguments, typically numeric.

Yes, that's really all.


=head1 OPTIONS

=over 6

=item B<-w> I<length>, B<--width> I<length>

Pad the displayed value with leading zeroes to ensure it satisfies a minimum length.


=item B<-b> I<string>, B<--before> I<string>

Prepend each <option> element with I<string>.


=item B<-a> I<string>, B<--after> I<string>

Append I<string> to each <option> element.


=item B<-n>, B<--newline>, B<--new-line>

Append a line-break character (I<U+000A>) after each <option>, as though B<--after>=$'\n' were entered in the arguments. Overrides B<--after>.


=item B<-i> I<num>, B<--indent> I<num>

Indent each <option> with I<num> tabs. Overrides B<--before>.


=item B<-v>, B<--verbatim>

Pad numeric values with zeroes, but don't use the unpadded value as a I<value> attribute in the <option> element.

=back


=head1 EXAMPLES

Generate a list of numbers representing the days of the month, with leading zeroes affixed where necessary:

	html-option-list -w 2 {1..31}

Do the same as above, but "pretty-print" it so it's not crapped on one frikkin' long line:

	html-option-list -w 2 --newline --indent=1 {1..31}

=cut



use warnings;
use strict;
use feature "say";
use Getopt::Long qw(:config auto_abbrev);


# Initialise output string
my $output	= "";


# Process command-line options
my $width	= 0;
my $before	= "";
my $after	= "";
my $newline	= 0;
my ($indent, $verbatim);
GetOptions("width=i" => \$width, "before=s" => \$before, "after=s" => \$after, "newline" => \$newline, "new-line" => \$newline, "indent=i" => \$indent, "verbatim" => \$verbatim);


# If --newline was enabled, set $after to... well, a new-line.
if($newline){ $after = "\n"; }


# Similarly, if $indent is specified, use its value as the number of tabs to set $before to.
if(defined($indent)){
	$before	=	"\t" x $indent;
}


# Define formatting string
my $format	= $before . ($width > 1 ?
	'<option' . ($verbatim ? '' : ' value="%1$s"') . '>%1$0'.$width.'s</option>' :
	'<option>%1$s</option>'
) . $after;



# Start formatting arguments.
while(@ARGV){
	$_	=	shift;
	printf $format, $_;
}
